### DISCUSSION TITLE
How to trigger menu list using Input component?

### DISCUSSION THREAD

**Discussion by ahmadxgani** [chakra-ui/chakra-ui#7407] [Q&A] [answered]
Hello,
is it possible to trigger the MenuList using the Input component? I am a little confused about that.

![image](https://user-images.githubusercontent.com/73988956/222115868-a2793e9f-4554-4510-803f-fec929158980.png)

**Comment by vitalijalbu** [comment]
I also want this, for an autocomplete

**Comment by carlosberti** [comment, marked as answer]
Not sure if that it is what you want, but you can attach the disclosures to the Menu element like this:

`<Menu isOpen={isOpen} onOpen={onOpen} onClose={onClose}>`

And then call the `onOpen()` at the input `onChange` or wherever you want. E.g.:

`<input onChange={() => { onOpen()}} />`

**Reply by ahmadxgani** [reply]
thanks for your help, it's worked! 👍

**Reply by ArthurGerbelot** [reply]
Can someone provide an example for this ? This is exactly what I need but cannot make it works, not sure to understand..

Tried something like that:
```
  let onOpen: () => void, onClose: () => void;
  return (
    <Menu onOpen={onOpen} onClose={onClose}>
    
      <Input onFocus={() => onOpen()} onBlur={() => onClose()} />

      <MenuList>
        <MenuItem>Download</MenuItem>
        <MenuItem>Example</MenuItem>
      </MenuList>
    </Menu >
  )
  ```

**Reply by carlosberti** [reply]
Try to do like this => 

`<Menu isOpen={isOpen} onOpen={onOpen} onClose={onClose}>`

**Reply by ArthurGerbelot** [reply]
Hum.... I did, this is the simplest version I tried but I got error `Cannot find name 'isOpen'. Did you mean 'open'?` (same for onOpen and onClose)

```
export function InputSearch(props: InputProps) {
  return (
    <Menu isOpen={isOpen} onOpen={onOpen} onClose={onClose}>

      <Input onFocus={() => onOpen()} onBlur={() => onClose()} />

      <MenuList>
        <MenuItem>Download</MenuItem>
        <MenuItem>Example</MenuItem>
      </MenuList>
    </Menu >
  )
}
```

**Reply by carlosberti** [reply]
You need either to import at the Input search component the useDisclosure from chakra or provide through props

`cont { isOpen, onOpen, onClose } = useDisclosure()`

**Reply by ArthurGerbelot** [reply]
Ho Wow okay ! Didn't know about that thx !!

**Comment by OCmilo** [comment]
I have the same use case but my problem is a little different. How do I make the menu open below my component? I tried combining MenuButton with the Input in multiple ways but I didn't work. The menu always opens at a non-specified place without it.

**Reply by carlosberti** [reply]
You can set a "placement" and a "gutter" at the `Menu` component. Take a look [here](https://chakra-ui.com/docs/components/menu/props)

**Reply by OCmilo** [reply]
The thing is that I think popper needs a reference of where to put it, and MenuButton has it. Placement and gutter are properties that suppose you have a referential.

**Reply by carlosberti** [reply]
I see. Maybe you could create a codepen with an example of what you want and send the link here

**Reply by austinbradshaw** [reply]
I build a quick sandbox trying to do the same thing with a Portal @OCmilo @carlosberti https://play.chakra-ui.com/playground?code=N4Igxg9gJgpiBcICWBbADhATgFwATAB0A7XXAIQFdtsIiAaY0gSSLSoZNwFkYiKPSPPgBkkAZ2wDuvCk2wwUUgApZsAQwA2UimJgARcWA0QxFTDA4BfXADNMEFLgDkAATAALNQGtMagLQUSAD05mpg2E7EqBg4uABKMGF4dg7OoeGRRMQwAB4xeJBEErgAgmhouAC8uAAUAJRVAHz4jLjm2GYkADwAojlq6BowuEGNxJbExDYUROFItLh9A2hD9S2chcXmNlXxieEAdDowCTb1rZt4wLi0APJovHS44vePN0QAwsa6uNbVxwYxEYTGYYOcNrQxBAhgdjABzGovB5EOqtdqdWqtUhdMacUjYlhsPDbSrAbbWFZhGDuaGwTCVAggQlURnvL5IMBeUl3ZHWUaTPH4rpCCjPKAMkAoACeItZSN4pPlRGsPIVwFVyrZ3xg3M+2ssuPxRtwXRUOE0uEK6iQRBgmFOpPJhuNRuFMlEEmdLtdIrkCkaItwAEYukFffIUF7vdjw-6AHIwADuuAA6jaoBBE6HY5GsdGTTnGq8SF8TDAoLgACpqABG2Zkftzgujbr4jaLyNwADEkEN622I1GXf2KB7sEOhUEzeoNBOR1HQ87UcriCBLEA
